src/python/pants/core/util_rules/partitions.py

```diff
@@ -39,13 +37,7 @@ class PartitionerType(Enum):
     """Registers a partitioner which returns the inputs as a single partition."""
 
 
-class PartitionKey(Protocol):
-    @property
-    def description(self) -> str:
```

Comment:
Putting it a different way from the PR description: I think this is an over-fit from the `lint` use-case that will be irritating when we try to reuse these types in the `test` case